src/wasm/wasm.cpp

```diff
@@ -1036,6 +1036,10 @@ void RefAs::finalize() {
     return;
   }
   switch (op) {
+    case RefAsNonNull:
+      // FIXME: when we support non-nullable types, switch to NonNullable
+      type = Type(value->type.getHeapType(), Nullable);
```

Comment:
What happens here if the `value` is unreachable?

Reply:
In that case the return on line 1036 would have been taken, and we'd never get here.

Reply:
aha, should have expanded the context 👍